src/main/java/sparksoniq/jsoniq/item/FunctionItem.java

```diff
@@ -52,24 +55,45 @@
     // signature contains type information for all parameters and the return value
     private FunctionSignature signature;
     private RuntimeIterator bodyIterator;
-    private Map<String, List<Item>> variablesInClosure;
+    private Map<String, List<Item>> _localVariablesInClosure;
```

Comment:
Could we encapsulate these three maps in a reusable class, that both function items and dynamic contexts can use? That may simplify the APIs.

Reply:
TODO add an issue but merge this

Reply:
@ghislainfourny Could you approve this PR?